src/codegen/mod.rs

```diff
@@ -232,55 +288,129 @@ fn gen_stmt(
                 _ => panic!("Currently ranges are assumed to be predeclared functions."),
             };
 
-            cu::Stmt::ForLoop {
+            CheckedStmt::Stmt(cu::Stmt::ForLoop {
                 init: Box::new(init),
-                cond,
+                cond,   // TODO needs some kind of checking
```

Comment:
Does the checking not happen before `cond` is generated as `cu::Stmt`?

Reply:
At the moment there is no checking at all. Just checking this condition is not enough. It needs to be checked before each iteration of the loop.  I am talking about a case like (no specific language):
```
int[] a = {1, 2, 3, 4};
for ( int i = 0; a[i] > 0 ; i++) {
    // code ...
}
```
This loop will have an out-of-bounds error for `i>=4`. 
We talked about not allowing conditions like the one above. That means not allowing indexing inside the condition of for-loops

Reply:
Makes sense to me. We can simply keep the TODO.